wagtail/admin/tests/pages/test_edit_page.py

```diff
@@ -1278,7 +1278,7 @@ def test_edit_after_change_language_code_without_revisions(self):
         )
 
     def test_page_edit_num_queries(self):
-        with self.assertNumQueries(40):
+        with self.assertNumQueries(46):
```

Comment:
With the above optimisation, we should be able to bring this to 41 (or maybe even keep it at 40 in this case).